### PR TITLE
Replace if/then with ternary in Math::ease

### PR DESCRIPTION
**Pull request by avivkeller**
This pull request replaces the if/then statements with ternary (mostly). It also makes the comments a bit longer for clarity.

### CODE REVIEW

**Comment by AThousandShips** [comment]
I consider these to be significantly worse for readability

**Comment by AThousandShips** [comment]
Adding a comment to compensate for worse readability sounds backwards, the readability isn't about if you can understand what it's supposed to do from the maths perspective, but that it's easy to break down the actual code

Adding comments might be good on its own, but these changes make the code worse IMO

File: core/math/math_funcs.cpp [hunk 1]
```diff
@@ -93,27 +93,18 @@ int Math::range_step_decimals(double p_step) {
 }
 
 double Math::ease(double p_x, double p_c) {
-	if (p_x < 0) {
-		p_x = 0;
-	} else if (p_x > 1.0) {
-		p_x = 1.0;
-	}
+	// Ensure that p_x stays within the range [0, 1]
+	p_x = (p_x < 0.0) ? 0.0 : ((p_x > 1.0) ? 1.0 : p_x);
```
Comment (AThousandShips) [inline core/math/math_funcs.cpp:97]: This is also doing redundant assignment, which is bad, it might be optimized out but it might not

Comment (RedYetiDev) [reply core/math/math_funcs.cpp:97]: The `p_x = ` assignment existed before the PR, how would you suggest it done? I'm happy to change it

Comment (AThousandShips) [reply core/math/math_funcs.cpp:97]: > Or do you mean that before-hand it only assigned when out-of-range?

Yes, that's the unnecessary assignment that didn't occur before

Comment (RedYetiDev) [reply core/math/math_funcs.cpp:97]: Ah, okay, I didn't really think of these little optimizations that were in place. Maybe your right and this PR has more cons thans pros. I can close if you'd like, unless you want a second opinion?

Comment (AThousandShips) [reply core/math/math_funcs.cpp:97]: It's up to you, at least this part should go as it hurts performance, but others might disagree

**Comment by RedYetiDev** [comment]
e

**Pull request closed without merging**